ui.apps/src/test/components/commerce/product/galleryTest.js

```diff
@@ -0,0 +1,172 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+'use strict';
+
+describe('Product', () => {
+
+    describe('Gallery', () => {
+
+        let galleryRoot;
+        let pageRoot;
+        let body;
+        let galleryItems = [
+            {
+                "path": "http://hostname.tld/image-a.jpg",
+                "label": "Image A"
+            },
+            {
+                "path": "http://hostname.tld/image-b.jpg",
+                "label": "Image B"
+            }
+        ];
+
+        before(() => {
+            body = document.querySelector('body');
+            pageRoot = document.createElement('div');
+            body.appendChild(pageRoot);
+        });
+
+        beforeEach(() => {
+            while (pageRoot.firstChild) {
+                pageRoot.removeChild(pageRoot.firstChild);
+            }
+            pageRoot.insertAdjacentHTML('afterbegin', `
+                <div data-gallery-role="galleryroot">
```

Comment:
We should extract this to a template or something.

Reply:
Are you referring to having the DOM snippets globally available through maybe `const` expressions or like handlebar templates?